File: java2csharp/java/Samsung/ProfessionalAudio.FloatingConotrller.WidgetsLibrary/MainDeviceLayout.java
package com.samsung.android.sdk.professionalaudio.widgets;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.widget.ImageButton;
import android.widget.LinearLayout;

import com.samsung.android.sdk.professionalaudio.app.SapaAppService;

import java.lang.ref.WeakReference;

class MainDeviceLayout extends DeviceActionsLayout {

    private int perpendicular;
    private int along;

    protected MainDeviceLayout(Context context, DeviceActionsLayoutData data,
            SapaAppService sapaAppService, boolean isExpanded, int orientation,
            ControlBar bar) {
        super(context, data, sapaAppService, isExpanded, orientation, bar);
    }

    @Override
    protected ImageButton createDeviceButton() {
        ImageButton button = (ImageButton)LayoutInflater.from(getContext()).inflate(R.layout.main_app_view, this,false);
        Drawable drawable = this.mData.mInstanceIcon;
        if(drawable.getIntrinsicHeight()> mDeviceIconMaxSize){
            drawable.setBounds(0,0,mDeviceIconMaxSize,mDeviceIconMaxSize);
        }
        button.setImageDrawable(drawable);
        button.setClickable(false);
        button.setLongClickable(true);

        Resources res = getResources();
        perpendicular = res.getDimensionPixelSize(R.dimen.default_controlbar_widget);
        along = res.getDimensionPixelSize(R.dimen.main_app_btn_length);
        return button;
    }

    protected LinearLayout.LayoutParams getAppBtnLayoutParams(int orientation){
        return orientation == LinearLayout.HORIZONTAL ?
                new LinearLayout.LayoutParams(along, perpendicular ) :
                new LinearLayout.LayoutParams(perpendicular , along);
    }
}

File: java2csharp/csharp/Samsung/ProfessionalAudio.FloatingConotrller.WidgetsLibrary/MainDeviceLayout.cs
namespace com.samsung.android.sdk.professionalaudio.widgets
{

	using Context = android.content.Context;
	using Resources = android.content.res.Resources;
	using Bitmap = android.graphics.Bitmap;
	using BitmapDrawable = android.graphics.drawable.BitmapDrawable;
	using Drawable = android.graphics.drawable.Drawable;
	using LayoutInflater = android.view.LayoutInflater;
	using ImageButton = android.widget.ImageButton;
	using LinearLayout = android.widget.LinearLayout;

	using SapaAppService = com.samsung.android.sdk.professionalaudio.app.SapaAppService;

	internal class MainDeviceLayout : DeviceActionsLayout
	{

		private int perpendicular;
		private int along;

		protected internal MainDeviceLayout(Context context, DeviceActionsLayoutData data, SapaAppService sapaAppService, bool isExpanded, int orientation, ControlBar bar) : base(context, data, sapaAppService, isExpanded, orientation, bar)
		{
		}

		protected internal override ImageButton createDeviceButton()
		{
			ImageButton button = (ImageButton)LayoutInflater.from(Context).inflate(R.layout.main_app_view, this,false);
			Drawable drawable = this.mData.mInstanceIcon;
			if (drawable.IntrinsicHeight > mDeviceIconMaxSize)
			{
				drawable.setBounds(0,0,mDeviceIconMaxSize,mDeviceIconMaxSize);
			}
			button.ImageDrawable = drawable;
			button.Clickable = false;
			button.LongClickable = true;

			Resources res = Resources;
			perpendicular = res.getDimensionPixelSize(R.dimen.default_controlbar_widget);
			along = res.getDimensionPixelSize(R.dimen.main_app_btn_length);
			return button;
		}

		protected internal override LinearLayout.LayoutParams getAppBtnLayoutParams(int orientation)
		{
			return orientation == LinearLayout.HORIZONTAL ? new LinearLayout.LayoutParams(along, perpendicular) : new LinearLayout.LayoutParams(perpendicular, along);
		}
	}

}